cfg/config.go

```diff
@@ -4,27 +4,88 @@ import (
 	"encoding/json"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
 	"github.com/muesli/beehive/bees"
+	gap "github.com/muesli/go-app-paths"
 )
 
+const appName = "beehive"
+const cfgFileName = "beehive.conf"
+
+// This can be easily replaced in tests so we can test without using
+// the real search paths
+var searchPaths = defaultSearchPaths
+
 // Config contains an entire configuration set for Beehive
 type Config struct {
 	Bees    []bees.BeeConfig
 	Actions []bees.Action
 	Chains  []bees.Chain
 }
 
+// DefaultPath returns Beehive's default config path
+//
+// The path returned is OS dependant. If there's an error
+// while trying to figure out the OS dependant path, "beehive.conf"
+// in the current working dir is returned.
+func DefaultPath() string {
+	userScope := gap.NewScope(gap.User, appName)
+	path, err := userScope.ConfigPath(cfgFileName)
+	if err != nil {
+		return cfgFileName
+	}
+
+	return path
+}
+
+func defaultSearchPaths() []string {
```

Comment:
You could use `gap`'s `LookupConfig` function for this. It returns all existing config files in the default (XDG) search paths.

Reply:
Totally missed that, will have a look.

Reply:
So using this as-is would introduce an additional complication, given that it'll also return system config paths even when using the user scope, if the file exists in `/etc/beehive/beehive.conf` for example.

If beehive is running as a regular user without write access to that `/etc` config file, we'd need to either:

* Panic and tell the user we can't write to the config file
* Teach beehive to gracefully degrade and operate in config read-only mode, including API and/or Web UI changes
* Write changes to the user config dir after reading the config from the system path, which could be a bit confusing for existing users perhaps.
* Remove system scope paths from the returned array, so only user paths `./beehive.conf` and `~/.config/beehive/beehive.conf` are returned.

I'm opting for the last one to keep things simple, but I don't have a strong preference here.